cpp/src/neighbors/scann/detail/scann_build.cuh

```diff
@@ -173,6 +173,9 @@ index<T, IdxT> build(
       res, kmeans_params, batch_view, raft::make_const_mdspan(centroids_view), batch_labels_view);
 
     dataset_vec_batches.prefetch_next_batch();
+
+    // Make sure work on device is finished before swapping buffers
+    raft::resource::sync_stream(res);
```

Comment:
prefetch_next_batch seems to be [synchronizing at the very end](https://github.com/rapidsai/cuvs/blob/46760d444567347a5ab132f6dd03246b85a0aee3/cpp/src/neighbors/detail/ann_utils.cuh#L550). Would it make better sense to have the sync_stream before prefetching the batch (above line 175)?

Reply:
Good point. I had not realized that prefetching is done using a different stream